### DISCUSSION TITLE
Missing bounced file and Received: field

### DISCUSSION THREAD

**Discussion by Metabaron1** [simple-login/app#610] [Self-hosting question] [answered]
Hi

I'm runnig self hosted 3.4.0. I have 2 questions:
- Most  bounced emails are saved in 2 files in upload/refused-emails directory. Could you tell me why some are missing and there is only full-....eml file, and body is lost.  Is there a way to keep all bodies? 
- And could you tell me why Received: field only contain the simple-login server.  It should also contain the full path before the simple-login server? It's a usefull  information when we use network filtering, for example to whitelist some IP.

Thanks

**Comment by nguyenkims** [comment, marked as answer]
> Most bounced emails are saved in 2 files in upload/refused-emails directory. Could you tell me why some are missing and there is only full-....eml file, and body is lost. Is there a way to keep all bodies?

Sometimes it's impossible to extract the original email because it isn't included in the bounce report so there's only "full-..." file in this case. This is due to some email services that don't respect the best practices of always including the bounced email in the bounce report.

> And could you tell me why Received: field only contain the simple-login server. It should also contain the full path before the simple-login server? It's a usefull information when we use network filtering, for example to whitelist some IP.

All existing headers are dropped before SimpleLogin forwards the email as some spam filters might not support SimpleLogin forwarding method which is different than a simple forward, SRS or ARC and they might block SimpleLogin email.

**Reply by Metabaron1** [reply]
OK I understand. Thanks for explanation.
